### PR TITLE
Move min airflow version to 2.3.0 for all providers

### PR DESCRIPTION
**Pull request by potiuk**
As of October 11 our providers are supposed to be compatible with Airflow 2.3+ and all code for backwards compatibility with Airflow 2.2 can be removed now.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -344,7 +344,7 @@ repos:
         # This check might be removed when min-airflow-version in providers is 2.2
       - id: check-airflow-2-2-compatibility
         name: Check that providers are 2.2 compatible.
-        entry: ./scripts/ci/pre_commit/pre_commit_check_2_2_compatibility.py
+        entry: ./scripts/ci/pre_commit/pre_commit_check_provider_airflow_compatibility.py
```
Comment (eladkal) [inline .pre-commit-config.yaml:365]: Need to change pre commit id and name

File: airflow/providers/jira/provider.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ versions:
   - 1.0.0
 
 dependencies:
-  - apache-airflow>=2.2.0
+  - apache-airflow>=2.3.0
```
Comment (eladkal) [inline airflow/providers/jira/provider.yaml]: I opened a PR to drop this provider

**Review by eladkal — APPROVED** [review]

**Comment by eladkal** [comment]
We need also update
https://github.com/apache/airflow#release-process-for-providers
With new date

File: .pre-commit-config.yaml (hunk 1 above)
Comment (potiuk) [reply .pre-commit-config.yaml:365]: Done.

File: airflow/providers/jira/provider.yaml (hunk 2 above)
Comment (potiuk) [reply airflow/providers/jira/provider.yaml]: I wille get to that PR shortly, but let's get it changed here in the meantime for consistency.

**Comment by potiuk** [comment]
> We need also update https://github.com/apache/airflow#release-process-for-providers With new date

Right. Updated.

**Comment by potiuk** [comment]
I'd love @uranusjr if you take a look as well. The biggest part of pre-2.3 was removal of dttm from "XCom.get_value" - mostly in operator links.

Also I am not sure if what I've done is correct. I assumed that after migration to 2.3 each task instance will have a "task_instance_key" and we will be able to use task instance key and `get_value` even for task instances created in previous Airflow versions (I believe migration takes care about it). 

I would love your comments here  @uranusjr if I understood correctly that we should genereally skip all the dttm usage and rely on task_instance_key being always present.

I have found however one case where it is not as straightforward and dttm is still used and I am not sure if we want to do anything about it:

```
class ExternalDagLink(BaseOperatorLink):
    """
    Operator link for ExternalTaskSensor and ExternalTaskMarker.
    It allows users to access DAG waited with ExternalTaskSensor or cleared by ExternalTaskMarker.
    """

    name = 'External DAG'

    def get_link(self, operator, dttm):
        ti = TaskInstance(task=operator, execution_date=dttm)
        operator.render_template_fields(ti.get_template_context())
        query = {"dag_id": operator.external_dag_id, "execution_date": dttm.isoformat()}
        return build_airflow_url_with_query(query)
```

In all the other cases I converted 'get_link" to 

```
    def get_link(
        self,
        operator: BaseOperator,
        *,
        ti_key: TaskInstanceKey,
```

**Comment by eladkal** [comment]
We also have:
https://github.com/apache/airflow/blob/b254a9f4bead4e5d4f74c633446da38550f8e0a1/airflow/providers/oracle/hooks/oracle.py#L428-L441

**Comment by potiuk** [comment]
> We also have:
> https://github.com/apache/airflow/blob/b254a9f4bead4e5d4f74c633446da38550f8e0a1/airflow/providers/oracle/hooks/oracle.py#L428-L441

Ah cool. Indeed.

**Comment by potiuk** [comment]
Actually, the DBApi one we could do before already after we split-out common.sql.

**Comment by potiuk** [comment]
Fixed.

**Comment by potiuk** [comment]
(also done the same for Trino, Snowflake and Databricks)

**Comment by eladkal** [comment]
Ahh we can also handle:
https://github.com/apache/airflow/blob/06acf40a4337759797f666d5bb27a5a393b74fed/airflow/providers/databricks/triggers/databricks.py#L30

https://github.com/apache/airflow/blob/06acf40a4337759797f666d5bb27a5a393b74fed/airflow/providers/google/cloud/triggers/cloud_composer.py#L31

**Review by Fokko — APPROVED** [review]

**Comment by potiuk** [comment]
This one also should cut the "compatibility" test by more than hour. Currently just installing latest providers on airflow 2.2 takes > 1 hour because `pip` tries to resolve dependencies and it tries very hard to do so (this is because of the complex dependency resolution caused by "old" airflow and "new" providers). This is -yet another reason why the 12 months compatibility for providers is a good idea.

**Comment by potiuk** [comment]
All right - seems I get it alll green - @uranusjr - is it possible you take a look at the XCom.get_value() changes before I merge this one?

**Comment by uranusjr** [comment]
Just did, I think you did it correctly.

**Comment by potiuk** [comment]
> Deferrable Operators only work starting Airflow 2.2

Indeed. Thanks @eladkal  for all the extra checks :)

**Pull request merged**